client/my-sites/plan-features-2023-grid/hooks/npm-ready/use-is-large-currency.ts

```diff
@@ -1,12 +1,55 @@
-import { getDisplayPrices } from '../../lib/get-display-prices';
-import type { GridPlan } from 'calypso/my-sites/plan-features-2023-grid/hooks/npm-ready/data-store/use-grid-plans';
+import { formatCurrency } from '@automattic/format-currency';
 
+const LARGE_ADD_ON_CURRENCY_CHAR_THRESHOLD = 7;
 const LARGE_CURRENCY_CHAR_THRESHOLD = 6;
 const LARGE_CURRENCY_COMBINED_CHAR_THRESHOLD = 9;
 
 interface Props {
-	gridPlans: GridPlan[];
-	returnMonthly?: boolean;
+	prices?: number[];
+	isAddOn?: boolean;
+	currencyCode: string;
+}
+
+function getDisplayPrices( prices: number[], currencyCode: string ) {
+	/**
+	 * Prices are represented in smallest units for a currency, and not as prices that
+	 * are actually displayed. Ex. $20 is the integer 2000, and not 20. To determine if
+	 * the display price is too long, we convert the integer to a display string.
+	 */
+	return prices.map( ( price ) =>
+		formatCurrency( price, currencyCode, {
+			stripZeros: true,
+			isSmallestUnit: true,
+		} )
+	);
+}
+
+function hasExceededPriceThreshold( displayPrices: string[], isAddOn = false ) {
```

Comment:
**Note:**

I've simplified the interface for `isLargeCurrency` quite a bit -- it now accepts a simple list of prices to evaluate.

In doing so, however, we add complexity to this helper `hasExceededPriceThreshold`. We were checking if the combined original and discounted plan prices exceeded 9 characters , because in the UI, the two prices are displayed side by side causing potential overflow.

Now we have to implicitly derive the information about the pairs of original and discounted prices. I think the tradeoff is worth it because it makes the consumer interface for `isLargeCurrency` noticeably more intuitive, but this is definitely up for discussion 😃